src/plugins/intel_gpu/src/graph/gather.cpp

```diff
@@ -30,8 +30,9 @@ layout gather_inst::calc_output_layout(gather_node const& node) {
         output_type = node.get_fused_output_layout().data_type;
     }
 
-    if (desc->batch_dim == 0) {
```

Comment:
I think that you might try to use Gather(data, indices, axis, batch_dims) with dummy data, indices, axis nodes. 
However lets go with current fix.